### ISSUE TITLE
Duplicate Messages in backfilled chats

### ISSUE THREAD

**Issue by Severin-S** [mautrix/instagram#73]
When backfilling chats, the first messages are duplicated. The exact number of duplications varies, I counted between 6 and 9 times the same messages. The number of messages duplicated also varies, I counted 3-6. All chats with duplicates have more than 5 messages, all chats without have less than 5 messages. This didn't occur yesterday with bride version `mautrix-instagram 0.2.3+dev.6556cbb7`.
I cleared the cache of Element Desktop (Windows) and also verified the existente of the duplicate messages in Element Android.
current bridge version: `mautrix-instagram 0.2.3+dev.6f40e012`.

I'm not sure what to search for in the logs, please tell me what information you need :)

Relevant config:
```
    max_startup_thread_sync_count: 40
    backfill:
        enable: true
        msc2716: true
        double_puppet_backfill: false
        max_conversations: -1
        min_sync_thread_delay: 5
        unread_hours_threshold: 0
        backoff:
            thread_list: 300
            message_history: 300
        incremental:
            max_pages: 10
            max_total_pages: -1
            page_delay: 5
            post_batch_delay: 20
```

**Comment by Severin-S** [comment]
I just noticed more weird behavior:
![ig_tmp](https://user-images.githubusercontent.com/26414182/208177492-30d3e136-8fc4-42e0-92bc-d1f297d6996e.png)
The first message in the chat is from the 10.1.19, the second and following are from the 17.1.19. The user that was invited 21 times is my puppet.

**Comment by sumnerevans** [comment]
Possibly fixed by d7ea410, but not sure.

**Comment by Severin-S** [comment]
I'm unable to reproduce it (still with the same version as when I opened the issue). After `delete-portal` the room is either recreated without duplicated messages or without historical messages at all.
